aten/src/ATen/core/ATenDispatch.h

```diff
@@ -40,17 +42,47 @@ static inline TensorTypeId dispatchTypeId(TensorTypeSet ts) {
 
 }
 
+namespace detail {
+  struct MultiDispatchTensorTypeSet : IterArgs<MultiDispatchTensorTypeSet> {
+    TensorTypeSet ts;
+    void operator()(const at::Tensor& x) {
+      ts = ts | x.type_set();
+    }
+    void operator()(TensorOptions x) {
```

Comment:
Why not also take this by const&?

Reply:
I suppose you could. TensorOptions is a pass by value object so it doesn't really matter one way or another.

Reply:
TensorOptions is quite a few words large, right? Why is it a "pass by value object"? For things above 1-2 words, passing by reference is usually faster.

Reply:
It's two words.

Reply:
```
// We should aspire to fit in one machine-size word; but a size greater than two
// words is too much.  (We are doing terribly on 32-bit archs, where we require
// three machine size words to store tensor options.  Eek!)
static_assert( sizeof(TensorOptions) <= sizeof(int64_t) * 2,
               "TensorOptions must fit in 128-bits" );
```